fdbserver/LogSystem.h

```diff
@@ -1087,13 +1088,30 @@ struct LogPushData : NonCopyable {
 		next_message_tags.clear();
 	}
 
-	Standalone<StringRef> getMessages(int loc) { return messagesWriter[loc].toValue(); }
+	Standalone<StringRef> getMessages(int loc) {
+		// Update written here because this is called less frequently.
+		Standalone<StringRef> value = messagesWriter[loc].toValue();
+		if (!written[loc]) {
+			BinaryWriter w(AssumeVersion(g_network->protocolVersion()));
+			Standalone<StringRef> v = w.toValue();
+			if (value.size() > v.size()) {
+				written[loc] = true;
+			}
+		}
+		return value;
+	}
+
+	float getEmptyLocationRatio() const {
+		auto count = std::count(written.begin(), written.end(), false);
+		return 1.0 * count / written.size();
```

Comment:
out of paranoia, can you assert or check `written.size() is > 0`? (I see its initialized before we call push)

Reply:
Done.